### ISSUE TITLE
Blocked: Failure to generate tunnel parameters: Failure to select a matching tunnel relay

### ISSUE THREAD

**Issue by myyc** [mullvad/mullvadvpn-app#5577]
## Issue report

Operating system: Linux 6.6.4

App version: 2023.6

### Issue description

i'm not an expert with mullvad but i had never encountered this issue before. i'm using the CLI.

```
mullvad relay set location de # any really
mullvad connect
mullvad status
```

the output is `Blocked: Failure to generate tunnel parameters: Failure to select a matching tunnel relay`. any clue?

**Comment by myyc** [comment]
`mullvad factory-reset` and reconnecting fixed this. mystery

**Comment by jasondunsmore** [comment]
Same here. Issuing a command:

    mullvad relay set hostname us-dal-ovpn-002

got me into a state where I was seeing the "Blocked: Failure to generate tunnel parameters: Failure to select a matching tunnel relay" error. No amount of logging out and back into my account would fix it. I only tried a factory reset when I saw this issue. It fixed the problem. I'm running version 2023.6.

**Comment by jasondunsmore** [comment]
BTW, "mullvad relay set hostname" was available in version 2023.4, but seems to have disappeared from 2023.6.

**Comment by raksooo** [comment]
This can happen if you select a server/location that is incompatible with your settings. Some examples of how this can happen:
* Set tunnel protocol to WireGuard and set location to an OpenVPN server
* Add filter to use only a specific provider or only Mullvad owned servers and then select a server that doesn't match those filters.

@jasondunsmore Now you can provide a hostname to `mullvad relay set location`. It accepts countries, cities and hostnames.
